web/platform/src/content/posts/Customer_Service_AI_Agent.md

```diff
@@ -0,0 +1,4312 @@
+                    print(
+                        f"✅ Query with context processed successfully. Response: {response[:50]}..."
+                    )
+                else:
+                    print(
+                        "⚠️ Skipping API test with context - set RUN_API_TESTS=true to run"
+                    )
+            except Exception as e:
+                print(f"❌ Context query test failed: {str(e)}")
+                return False
+
+            print("\nAll tests completed successfully! ✅")
+            return True
+
+        # Run the tests
+        success = run_tests()
+        sys.exit(0 if success else 1)
+
+    ```
+    </div>
+
+- `run_assistant.sh -`
+    <div style="max-height: 250px; max-width: 50vw; overflow: auto;">
+
+    ```bash
+    #!/bin/bash
+
+    # Get the directory where this script is located
+    DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+    echo "Starting Hybrid Agent test from directory: $DIR"
+    echo "Python version:"
+    python3 --version
+
+    # Run the Python script
+    echo "Running Python script ..."
+    python3 "$DIR/assistant.py"
+
+    # Capture the exit code
+    exit_code=$?
+
+    echo "Python script completed with exit code: $exit_code"
+
+    # Return the same exit code
+    exit $exit_code
+
+    ```
+    </div>
+
+- `BUILD file -`
+    <div style="max-height: 250px; max-width: 50vw; overflow: auto;">
+
+    ```bash
+    load("@rules_python//python:py_binary.bzl", "py_binary")
+
+    py_binary(
+        name="hybrid_agent",
+        srcs=["assistant.py"],
+        main="assistant.py",
+        deps=[
+            "@pypi//python_dotenv",
+            "//src/models/claude_client:claude_client",
+            "//src/models/huggingface:huggingface_client",
+        ],
+        data=[
+            "//:.env",  # Moved from deps to data
+        ],
+        imports=[
+            ".",
+            "../..",  # For imports starting with 'src'
+        ],
+        visibility=["//visibility:public"],
+    )
+
+    `sh_test`(
+        name="hybrid_agent_test",
+        srcs=["run_assistant.sh"],
+        data=[
+            ":hybrid_agent",
+        ],
+    )
+
+    ```
+    </div>
+
+To see an open-source model come up with vector embeddings for context documents that are then sent to Claude to get an MCP compliant response, run `assistant.py`.
+
+For local execution, use: <br>
+`bazel run //src/agents/hybrid_assistant:hybrid_agent`
+
+For remote execution, use: <br>
+`bazel test //src/agents/hybrid_assistant:hybrid_agent_test`
+
+## **5) Training / Fine-Tuning On NativeLink Cloud**
+
+One of the key advantages of our setup is the ability to leverage powerful cloud resources through NativeLink's remote execution capabilities. This demo uses CPUs for fine-tuning but we've Nvidia GPUs in our infrastructure to offer; contact us to know more. Let's create a script that demonstrates how to run training jobs on remote hardware:
```

Comment:
You don't need to say `we've in our infrastructure to offer` nor  `contact us to know more`.

You can simply say that this tutorial uses CPUs to keep costs down, but NativeLink works with GPUs as well. No need to sell.